### PR TITLE
Bluetooth: Add Kconfig template for log inheriting

### PR DESCRIPTION
**Pull request by theob-pro**
Add a new Kconfig template that allow log modules to inherit their log level from their parent module.
For example, the logs used in the Bluetooth audio like `BT_AUDIO_STREAM_LOG_LEVEL` can inherit their level from `BT_AUDIO_LOG_LEVEL`.

Update the Bluetooth specific Kconfig legacy log template to use the new template.

Also, remove unnecessary conditionnal definition of `bt_tbs_dbg_print_calls` in `subsys/bluetooth/audio/tbs.h`.

### CODE REVIEW

File: subsys/bluetooth/audio/tbs.c [hunk 1]
```diff
@@ -2430,7 +2430,6 @@ void bt_tbs_register_cb(struct bt_tbs_cb *cbs)
 	tbs_cbs = cbs;
 }
 
-#if defined(CONFIG_BT_TBS_LOG_LEVEL_DBG)
```
Comment (jori-nordic) [inline subsys/bluetooth/audio/tbs.c:2433]: the linker will remove this, right @Thalley ?

Comment (Thalley) [reply subsys/bluetooth/audio/tbs.c:2433]: If it won't be used, yes. 

The main point of the guard was to ensure that it wasn't available when debug wasn't enabled. 

The guard can be removed from the header file as you've done (it should in any case). 

We should still be able to guard the implementation with `#if defined(CONFIG_BT_TBS_LOG_LEVEL_DBG)` - Doesn't that work?

**Review by Thalley — DISMISSED** [review]

**Review by szymon-czapracki — DISMISSED** [review]

**Comment by jhedberg** [comment]
The changes look fine to me, but I think it would have been cleaner to split the logging subsystem changes into a separate commit before the Bluetooth changes (still part of this same PR, however).

**Comment by jhedberg** [comment]
> Also, remove unnecessary conditionnal definition of bt_tbs_dbg_print_calls in subsys/bluetooth/audio/tbs.h and the implementation in the corresponding .c.

Is this part tightly related to the others? E.g. will the build fail if the other changes are taken but not this change? If not, then this should also be split into a separate commit.

**Review by nordic-krch — DISMISSED** [review]

**Comment by theob-pro** [comment]
Rebased to split modification into different commit as suggested by @jhedberg.

**Comment by theob-pro** [comment]
@jhedberg Okay, I was not sure if I should follow the prefix of the PR or not. I will update it.

**Comment by theob-pro** [comment]
Reworded a commit title as suggested by @jhedberg.
Removed redundant code in the legacy template.

**Review by alwa-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**